pkg/provision/migrations_202203_rbac_resource_fix.go

```diff
@@ -302,7 +300,23 @@ func migratePost202203RbacRule(r *rbac.Rule, rx *resIndex) (op actionType) {
 					return actionRemove
 				}
 			} else {
-				return actionRemove
+				// fetch record from store based on each module,
+				// 		to avoid fetching of all record of all module
+				for _, mod := range rx.modules {
+					if validID(p1) && validID(p2) {
+						continue
+					}
+					rec, err := store.LookupComposeRecordByID(ctx, s, mod, ID)
+					if err != nil && validID(rec.ModuleID) && validID(rec.NamespaceID) {
+						rx.records[rec.ID] = rec
```

Comment:
From what I see `rx.records` no longer does anything so you could just remove it?